include/jemalloc/internal/extent_inlines.h

```diff
@@ -211,6 +220,12 @@ extent_szind_set(extent_t *extent, szind_t szind) {
 }
 
 JEMALLOC_INLINE void
+extent_nfree_set(extent_t *extent, unsigned nfree) {
```

Comment:
nit: instead of set, maybe provide an extent_init (which sets nfree), and extent_nfree_inc / _dec, which adds / subs 1 << EXTENT_BITS_NFREE_SHIFT. Or even making nfree the lowest bits if you feel it's worth the trouble.